experimental/dds/tree2/src/shared-tree/sharedTreeEditBuilder.ts

```diff
@@ -0,0 +1,53 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { ChangeFamilyEditor, TreeStoredSchema } from "../core";
+import { DefaultEditBuilder, IDefaultEditBuilder, ModularChangeFamily } from "../feature-libraries";
+import { SharedTreeChange } from "./sharedTreeChangeTypes";
+
+/**
+ * SharedTree editor for transactional tree data and schema changes.
+ * @alpha
+ */
+export interface ISharedTreeEditor extends IDefaultEditBuilder {
+	/**
+	 * Updates the stored schema.
+	 * @param oldSchema - The schema being overwritten.
+	 * @param newSchema - The new schema to apply.
+	 * @alpha
+	 */
+	setStoredSchema(oldSchema: TreeStoredSchema, newSchema: TreeStoredSchema): void;
```

Comment:
I see you opted against the design we had discussed last where this would have exposed a `schema` getter that returned an object with all of schema's editing methods (currenly only `set` but I immagine that could change). I'm curious why.

Reply:
Done!